### PR TITLE
feat: [IOBP-313] New wallet payment pick PSP

### PR DESCRIPTION
**Pull request by Hantex9**
## Short description
This PR implements the pick PSP screen showing a list of PSPs available indicating the amount fee. The user can sort the list by default (showing the flagged `onUs` as first and sorting by amount ), by PSP name and amount;

## List of changes proposed in this pull request
- Upgraded design-system library version to the latest one to support the `RadioGroup` with the list item radio
- Adapted the UI of `WalletPaymentPickPspScreen`;
- Created two new actions to sort a PSP and reset a PSP selected;
- Edited the handle of the PSP saga, automatically picking the PSP with the `onUs` flag to true;
- Created a custom bottom sheet hook to show the available sorting modes;

## How to test
Go into the new wallet payment playground flow and navigate forward to the PSP selection

## Preview

https://github.com/pagopa/io-app/assets/34343582/fe003285-d832-4d1a-8535-ccdb82002ed9

### CODE REVIEW

File: ts/features/walletV3/payment/screens/WalletPaymentPickPspScreen.tsx [hunk 1]
```diff
@@ -56,44 +83,120 @@ const WalletPaymentPickPspScreen = () => {
...
+        accessibilityLabel: I18n.t("wallet.payment.psp.continueButton"),
+        onPress: handleContinue,
+        disabled: isLoading || !canContinue,
+        loading: isLoading
+      }}
+    >
+      <SelectPspHeadingContent />
+      {!isLoading &&
+        pspList?.map((psp, index) => (
+          <>
```
Comment (mastro993) [inline ts/features/walletV3/payment/screens/WalletPaymentPickPspScreen.tsx:180]: Each item should have an unique key. What about this?
```suggestion
          <React.Fragment key={`psp_${index}`}>
```

File: ts/features/walletV3/payment/screens/WalletPaymentPickPspScreen.tsx [hunk 2]
```diff
@@ -56,44 +83,120 @@ const WalletPaymentPickPspScreen = () => {
...
 
+  const sortButtonProps: ListItemHeader["endElement"] = React.useMemo(
+    () => ({
+      type: "buttonLink",
+      componentProps: {
+        label: I18n.t("wallet.payment.psp.pspSortButton"),
+        accessibilityLabel: I18n.t("wallet.payment.psp.pspSortButton"),
+        onPress: () => {
+          present();
+        }
```
Comment (mastro993) [inline ts/features/walletV3/payment/screens/WalletPaymentPickPspScreen.tsx:128]:
```suggestion
        onPress: present
```

File: ts/features/walletV3/payment/screens/WalletPaymentPickPspScreen.tsx [hunk 3]
```diff
@@ -56,44 +83,120 @@ const WalletPaymentPickPspScreen = () => {
...
+  React.useEffect(
+    () => () => {
+      dispatch(walletPaymentResetPickedPsp());
+    },
+    [dispatch]
+  );
+
+  const handleChangePspSorting = (sortType: WalletPaymentPspSortType) => {
+    setShowFeaturedPsp(sortType === "default");
+    dispatch(walletPaymentSortPsp(sortType));
```
Comment (mastro993) [inline ts/features/walletV3/payment/screens/WalletPaymentPickPspScreen.tsx:95]: The sort should be made on the UI, and not on data layer.
We should manage this with a `sortedPspList` directly in the screen component.

**Review by mastro993 — CHANGES_REQUESTED** [review]

File: ts/features/walletV3/payment/screens/WalletPaymentPickPspScreen.tsx (hunk 1 above)
Comment (Hantex9) [reply ts/features/walletV3/payment/screens/WalletPaymentPickPspScreen.tsx]: This logic has been removed in favor of `RadioGroup` integration and has been addressed in the following commit: 487939547b2f2447a9e68139c288641a345a67ef

File: ts/features/walletV3/payment/screens/WalletPaymentPickPspScreen.tsx (hunk 3 above)
Comment (Hantex9) [reply ts/features/walletV3/payment/screens/WalletPaymentPickPspScreen.tsx]: I agree! Addressed into 487939547b2f2447a9e68139c288641a345a67ef 👍

File: ts/features/walletV3/payment/screens/WalletPaymentPickPspScreen.tsx (hunk 2 above)
Comment (Hantex9) [reply ts/features/walletV3/payment/screens/WalletPaymentPickPspScreen.tsx]: Addressed into 487939547b2f2447a9e68139c288641a345a67ef

File: ts/features/walletV3/payment/screens/WalletPaymentPickPspScreen.tsx [hunk 4]
```diff
@@ -56,46 +89,128 @@ const WalletPaymentPickPspScreen = () => {
...
+        label: psp.bundleName ?? I18n.t("wallet.payment.psp.defaultName"),
+        isSuggested: psp.onUs && showFeaturedPsp,
+        suggestReason: I18n.t("wallet.payment.psp.featuredReason"),
+        formattedAmountString: formatNumberCentsToAmount(
+          psp.taxPayerFee || 0,
+          true,
+          "right"
+        )
+      }));
+
```
Comment (mastro993) [inline ts/features/walletV3/payment/screens/WalletPaymentPickPspScreen.tsx:214]: This could be simplified using `fp-ts`, what do you think??

**Review by mastro993 — CHANGES_REQUESTED** [review]

File: ts/features/walletV3/payment/screens/WalletPaymentPickPspScreen.tsx (hunk 4 above)
Comment (Hantex9) [reply ts/features/walletV3/payment/screens/WalletPaymentPickPspScreen.tsx]: Addressed into 8f9bd167bab8c4e8883c76eee21bdfd339c9ddc6, let me know what do you think about it 👍

**Review by mastro993 — APPROVED** [review]
LGTM

**Comment by CrisTofani** [comment]
@Hantex9 @mastro993 I just pushed a fix commit [4f53493](https://github.com/pagopa/io-app/pull/5302/commits/4f53493fdcea2167be5bfe8ffc0b938e1829b5b2) to solve the compilation error in `useHeaderProps` hook

**Pull request merged**